trustymail/trustymail.py

```diff
@@ -221,20 +215,19 @@ def spf_scan(domain):
                 logging.debug("\tResult Differs: Expected [{0}] - Actual [{1}]".format(result, response[0]))
                 domain.errors.append("Result Differs: Expected [{0}] - Actual [{1}]".format(result, response[0]))
 
-    except DNSError as error:
+    except (dns.resolver.NoAnswer, dns.exception.Timeout, dns.resolver.NXDOMAIN) as error:
         handle_error("[SPF]", domain, error)
 
 
 def dmarc_scan(domain):
     # dmarc records are kept in TXT records for _dmarc.domain_name.
     try:
         dmarc_domain = '_dmarc.%s' % domain.domain_name
-        for record in dnslookup(dmarc_domain, 'TXT'):
-
-            record_text = record_to_str(record)
+        for record in dns.resolver.query(dmarc_domain, 'TXT'):
+            record_text = record.to_text()
 
             if record_text.startswith("\""):
-                record_text = record[1:-1]
+                record_text = record_text[1:-1]
```

Comment:
Does this (`record` -> `record_text`) change the logical behavior of this function? or just something that is different about the two DNS libraries?

Reply:
Also, I wonder if rather than the conditional, it would be better to simplify to:

```python
record_text = record_text.lstrip('"')
```

Reply:
First question:  The new library returns an object.  If it's an MX record then you have to access the `exchange` and `preference` member variables.  If you treat the object as a string you get something like `10 mail.blah.com`.

Second question:  I think you're right, and I'll make the change.